docs/source/installation.mdx

```diff
@@ -0,0 +1,139 @@
+# Installation
+
+Before you start, you will need to setup your environment by installing the appropriate packages.
+
+🤗 `huggingface_hub` is tested on **Python 3.7+**.
+
+## Install with pip
+
+It is highly recommended to install 🤗 `huggingface_hub` in a [virtual environment](https://docs.python.org/3/library/venv.html).
+If you are unfamiliar with Python virtual environments, take a look at this [guide](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/).
+A virtual environment makes it easier to manage different projects, and avoid compatibility issues between dependencies.
+
+Start by creating a virtual environment in your project directory:
+
+```bash
+python -m venv .env
+```
+
+Activate the virtual environment. On Linux and MacOs:
+
+```bash
+source .env/bin/activate
+```
+
+Activate Virtual environment on Windows
+
+```bash
+.env/Scripts/activate
+```
+
+Now you're ready to install 🤗 `huggingface_hub` [from the PyPi registry](https://pypi.org/project/huggingface-hub/):
+
+```bash
+pip install --upgrade huggingface_hub
+```
+
+Once done, [check installation](#check-installation) is working correctly.
+
+### Install optional dependencies
+
+Some dependencies of `huggingface_hub` are [optional](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies).
+When not installed, some `huggingface_hub` features might be disabled or degraded. Those dependencies
+are optional as they are not required to run the core features of `huggingface_hub`.
+
+You can install optional dependencies via `pip`:
+```bash
+# Install dependencies for tensorflow-specific features
+# /!\ Warning: this is not equivalent to `pip install tensorflow`
+pip install 'huggingface_hub[tensorflow]'
+
+# Install dependencies for both torch-specific and CLI-specific features.
+pip install 'huggingface_hub[cli,torch]'
+```
+
+Here is the list of optional dependencies in `huggingface_hub`:
+- `cli`: provide a more convenient CLI interface. `huggingface-cli` can still be run without these.
```

Comment:
I would consider not mentioning the `huggingface-cli` to avoid confusion

```suggestion
- `cli`: provide a more convenient CLI interface for `huggingface_hub`.
```